lib/cretonne/src/ir/entities.rs

```diff
@@ -55,6 +55,12 @@ impl Value {
     }
 }
 
+impl Default for Value {
```

Comment:
This is unfortunate, I would prefer if we could avoid defining `Default` for `Value`.

Note that `EntityMap` doesn't actually require its elements to implement `Default` because it already stores the default element to use in the `default` member.

We should add a `EntityMap::with_default(default: V)` constructor in addition to `EntityMap::new()` which does require `V: Default`. Using such a method, you should be able to provide default values even for your nested mappings, without requiring anything to implement `Default`.

Reply:
Makes sense. I added a patch to do that now.